service/src/main/java/org/mskcc/cmo/metadb/service/impl/PatientCorrectionHandlingServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+package org.mskcc.cmo.metadb.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Message;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.mskcc.cmo.metadb.model.MetadbPatient;
+import org.mskcc.cmo.metadb.model.MetadbSample;
+import org.mskcc.cmo.metadb.model.SampleMetadata;
+import org.mskcc.cmo.metadb.service.CrdbMappingService;
+import org.mskcc.cmo.metadb.service.MetadbPatientService;
+import org.mskcc.cmo.metadb.service.MetadbSampleService;
+import org.mskcc.cmo.metadb.service.PatientCorrectionHandlingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PatientCorrectionHandlingServiceImpl implements PatientCorrectionHandlingService {
+    @Value("${metadb.correct_cmoptid_topic}")
+    private String CORRECT_CMOPTID_TOPIC;
+
+    @Value("${request_reply.cmo_label_generator_topic}")
+    private String CMO_LABEL_GENERATOR_REQREPLY_TOPIC;
+
+    @Value("${num.new_request_handler_threads}")
+    private int NUM_NEW_REQUEST_HANDLERS;
+
+    @Autowired
+    private CrdbMappingService crdbMappingService;
+
+    @Autowired
+    private MetadbPatientService patientService;
+
+    @Autowired
+    private MetadbSampleService sampleService;
+
+    private static Gateway messagingGateway;
+    private static final Log LOG = LogFactory.getLog(PatientCorrectionHandlingServiceImpl.class);
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private static boolean initialized = false;
+    private static volatile boolean shutdownInitiated;
+    private static final ExecutorService exec = Executors.newCachedThreadPool();
+    private static final BlockingQueue<Map<String, String>> correctCmoPatientIdQueue =
+            new LinkedBlockingQueue<Map<String, String>>();
+    private static CountDownLatch correctCmoPatientIdShutdownLatch;
+
+    @Override
+    public void initialize(Gateway gateway) throws Exception {
+        if (!initialized) {
+            messagingGateway = gateway;
+            setupCorrectCmoPatientIdHandler(messagingGateway, this);
+            initializeNewMessageHandlers();
```

Comment:
let's rename this method in all the classes to something like `initializeMessageHandlers()`